src/6_branch.js

```diff
@@ -1965,3 +1965,36 @@ Branch.prototype['trackCommerceEvent'] = wrap(callback_params.CALLBACK_ERR, func
 	});
 	done();
 });
+/*
+ logEvent() documentation
+ */
+Branch.prototype['logEvent'] = wrap(callback_params.CALLBACK_ERR, function(done, name, eventAndCustomData, contentItems) {
+	name = name && utils.validateParameterType(name, 'string') ? name : null;
+	eventAndCustomData = eventAndCustomData && utils.validateParameterType(eventAndCustomData, 'object') ? eventAndCustomData : null;
+	contentItems = contentItems && utils.validateParameterType(contentItems, 'array') ? contentItems : null;
+
+	var endpoint = utils.determineV2EventType(name);
+	var extractedEventAndCustomData = utils.separateEventAndCustomData(eventAndCustomData);
+
+	var eventCallback = function(err, data) {
+		return done(err || null);
+	};
+
+	if (endpoint === 'STANDARD') {
+		this._api(resources.logStandardEvent, {
+			"name": name,
+			"custom_data": safejson.stringify(extractedEventAndCustomData && extractedEventAndCustomData['custom_data'] || {}),
+			"event_data": safejson.stringify(extractedEventAndCustomData && extractedEventAndCustomData["event_data"] || {}),
+			"content_items": safejson.stringify(contentItems || [])
+		}, eventCallback);
+	}
+	else {
+		this._api(resources.logCustomEvent, {
+			"name": name,
+			"custom_data": safejson.stringify(extractedEventAndCustomData && extractedEventAndCustomData["custom_data"] || {}),
+			"event_data": safejson.stringify(extractedEventAndCustomData && extractedEventAndCustomData['event_data'] || {}),
+			"content_items": safejson.stringify(contentItems || [])
+		}, eventCallback);
+	}
```

Comment:
Just the endpoints are different so I've created some logic to determine which endpoint to route the request to. I was doing what I just noted earlier but saw some warnings when running make. I can't reproduce those warnings so I reverted to the previous approach.